### ISSUE TITLE
Search failed based on "url"

### ISSUE THREAD

**Issue by JoeLiu996** [typesense/typesense#1037]
## Description

I used typesense for indexing my document and do vector search. Currently I encounter a bug where I cannot search document based on a field called 'url'.
This is the code which cause bug. The bug is that I cannot get the expected document from this query even though I know the document is inside the collection.
```
def search_url(url):
    search_params = {
        'q': url,
        'query_by': 'url',
        'per_page': 1
    }
    response = typesense_client.collections['embedding_output_test'].documents.search(search_params)
    #print(response)
    if response['found'] == 0:
        print("documents not found")
        return None
    return response['hits'][0]['document']
```
## Steps to reproduce

Please use this bash script to reproduce the issue.
```
export TYPESENSE_API_KEY=xyz

curl "https://4t2dohp5vegxk7jnp-1.a1.typesense.net:443/collections" \
       -X POST \
       -H "Content-Type: application/json" \
       -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
       -d '{
         "name": "faq",
         "fields": [
           {"name": "url", "type": "string"}
         ]
       }'
       
curl "https://4t2dohp5vegxk7jnp-1.a1.typesense.net:443/collections/faq/documents/import?action=create" \
        -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
        -H "Content-Type: text/plain" \
        -X POST \
        -d '{"url": "https://www.cpf.gov.sg/member/infohub/cpf-clarifies/policy-faqs/why-cant-i-do-a-cpf-transfer-to-extended-family"}
            {"url": "https://www.cpf.gov.sg/member/infohub/cpf-clarifies/policy-faqs/why-interest-earned-on-cpf-life-premium-not-paid-to-beneficiaries"}'
            
curl "https://4t2dohp5vegxk7jnp-1.a1.typesense.net:443/multi_search" \
        -X POST \
        -H "Content-Type: application/json" \
        -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
        -d '{
          "searches": [
            {
              "collection": "faq",
              "q": "https://www.cpf.gov.sg/member/infohub/cpf-clarifies/policy-faqs/why-interest-earned-on-cpf-life-premium-not-paid-to-beneficiaries",
              "query_by": "url"
            }
          ]
        }'
```

## Expected Behavior
The document is actually inside the collection, because if I retrieve all documents and then do the search, can find the corresponding document:
```
def another_way(url):
    documents = typesense_client.collections['embedding_output_test'].documents.export()
    curr_json = [json.loads(document) for document in documents.splitlines()]
    matched_entry = None
    for document in curr_json:
        if document["url"] == url:
            matched_entry = document
            break
    print(matched_entry["url"])
    return matched_entry
```

## Actual Behavior
For first url in bash script, can search the result.

But for second url in bash script, the response is
```
{'facet_counts': [], 'found': 0, 'hits': [], 'out_of': 341, 'page': 1, 'request_params': {'collection_name': 'embedding_output_test', 'per_page': 1, 'q': '[https://www.cpf.gov.sg/member/infohub/cpf-clarifies/policy-faqs/why-interest-earned-on-cpf-life-premium-not-paid-to-beneficiaries'](https://www.cpf.gov.sg/member/infohub/cpf-clarifies/policy-faqs/why-interest-earned-on-cpf-life-premium-not-paid-to-beneficiaries%27)}, 'search_cutoff': False, 'search_time_ms': 0}
```

**Comment by jasonbosco** [comment]
@JoeLiu996 Thank you for opening this detailed report!

By default Typesense removes all special characters when indexing strings. 

So the following URL:

```
https://www.cpf.gov.sg/member/infohub/cpf-clarifies/policy-faqs/why-interest-earned-on-cpf-life-premium-not-paid-to-beneficiaries
```

will get converted to one long word:

```
httpswwwcpfgovsgmemberinfohubcpfclarifiespolicyfaqswhyinterestearnedoncpflifepremiumnotpaidtobeneficiaries
```

Typesense will then index the first 100 _characters_ of this long word as: 

```
httpswwwcpfgovsgmemberinfohubcpfclarifiespolicyfaqswhyinterestearnedoncpflifepremiumnotpaidtobenefic
```

Now at search time, Typesense also removes special characters. But we currently don't apply this 100 characters limit to words in the `q` string - this is a bug. So Typesense is trying to match the full untruncated word to the truncated word in the index and so no match is found. 

We'll fix this bug. 

But in the meantime a good workaround, especially in the case of long strings like URLs, is to add url special chars to `token_separators` like this:

```json
{
  "name": "faq",
  "fields": [
    {"name": "url", "type": "string"}
  ],
  "token_separators": ["/", ".", "-"]
}
```

This will cause Typesense to index the URL above as multiple distinct words like this:

```
www
cpf
gov
sg
member
infohub
cpf
clarifies
policy
faqs
why
interest
earned
on
cpf
life
premium
not
paid
to
beneficiaries
```

Since these words are much smaller than 100 characters, they'll all get indexed as is and get matched when you use the same query. 

I'll leave this issue open while we fix the bug.

**Comment by kishorenc** [comment]
This is fixed in `:0.25.0.rc35`.
